File: Project/mobile/android/app/src/main/java/com/bookease/app/ui/screens/home/HomeViewModel.kt
package com.bookease.app.ui.screens.home

import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import androidx.lifecycle.ViewModel
import com.bookease.app.data.mock.MockData
import com.bookease.app.data.models.BookCategory
import com.bookease.app.data.models.Booking
import com.bookease.app.data.models.BookingStatus
import com.bookease.app.data.models.Service

class HomeViewModel : ViewModel() {

    var searchText by mutableStateOf("")
        private set

    val user = MockData.currentUser

    val upcomingBooking: Booking? = MockData.bookings
        .filter { it.status == BookingStatus.CONFIRMED }
        .minByOrNull { it.startsAt }

    val popularServices: List<Service> = MockData.services.take(4)

    val categories: List<BookCategory> = MockData.categories

    fun onSearchChange(text: String) { searchText = text }
}

File: Project/mobile/iOS/BookEase/Features/Home/HomeViewModel.swift
import Foundation
import Observation

@Observable
final class HomeViewModel {
    var upcomingBooking: Booking? = MockData.bookings.first { $0.status == .confirmed }
    var popularServices: [Service] = Array(MockData.services.prefix(4))
    var categories: [BookCategory] = MockData.categories
    var searchText: String = ""
}
